includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1785,6 +1775,38 @@ public function unlock_order_payment( $order ) {
 		$order->save_meta_data();
 	}
 
+	/**
+	 * Retrieves the existing lock for an order.
+	 *
+	 * @param WC_Order $order The order to retrieve the lock for
+	 * @return mixed
+	 */
+	protected function get_order_existing_lock( $order ) {
+		$order->read_meta_data( true );
+		return $order->get_meta( '_stripe_lock_payment', true );
+	}
+
+	/**
+	 * Checks if an order is locked for payment processing.
+	 *
+	 * @param WC_Order $order The order to check the lock for
+	 * @return bool
+	 */
+	protected function is_order_payment_locked( $order ) {
+		$existing_lock = $this->get_order_existing_lock( $order );
+		if ( $existing_lock ) {
+			$parts      = explode( '|', $existing_lock ); // Format is: "{expiry_timestamp}" or "{expiry_timestamp}|{pi_xxxx}" if an intent is passed.
+			$expiration = (int) $parts[0];
+
+			// If the lock is still active, return true.
+			if ( time() <= $expiration ) {
```

Comment:
Just wanted to confirm, is the intent-related check intentionally removed?

Reply:
Not really, it got lost during the change in the implementation. Thanks for flagging. But I attempted to add it back to see how it would behave, and it does not work correctly due to the lack of payment intent at the beginning of the payment flow. I removed it for good now, along with the remaining references.